src/cdomains/floatDomain.ml

```diff
@@ -328,79 +343,89 @@ module FloatIntervalImpl(Float_t : CFloatType) = struct
   let unknown_IInt = IntDomain.IntDomTuple.top_of IInt
 
   let isfinite op =
-    match op with 
-    | Some v -> true_nonZero_IInt
-    | None -> unknown_IInt
+    match op with
```

Comment:
In order to avoid this pattern matching on `Bot, Top, Interval` over and over again, we should introduce a method similar to `eval_binop` - maybe `eval_unop`.
For the `isinf` etc. you can also add an optional parameter on what should happen with `Top` and default it to `top ()` for other invocations.